mautic/mautic
# Fix datetime segment filter empty expression

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
If you use datettime segment filter and use empty, then edit page return 

```
Internal Server Error - PHP Notice - Trying to access array offset on value of type null 
/app/bundles/LeadBundle/Form/DataTransformer/FieldFilterTransformer.php:58 at
```



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Crate segment with datetime and choose empty expression
3. Save segment and try edit it
4. You should see error on dev enviroment

![image](https://user-images.githubusercontent.com/462477/151519612-c88f9197-f963-42bb-8dce-a40257715320.png)





<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10806"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
